<?xml version="1.0"?>
<!--
Launch the path follower with the inputscaling controller for a car-like (Ackermann) robot.
-->
<launch>
    <group ns="path_follower">
        <group ns="controller">
            <!-- parameters, that are dependent of the controller or the robot model -->
            <group ns="ackermann_inputscaling">
                <!-- simulation -->
                <param name="k_forward" value="4.5" />
                <param name="k_backward" value="10.0" />
                <param name="factor_k1" value="1.0" />
                <param name="factor_k2" value="3.0" />
                <param name="factor_k3" value="3.0" />
                <param name="factor_steering_angle" value="1.0" />

                <!-- robot -->
                <!--param name="k_forward" value="4.5" />
                <param name="k_backward" value="7.0" />
                <param name="factor_k1" value="1.0" />
                <param name="factor_k2" value="3.0" />
                <param name="factor_k3" value="3.0" />
                <param name="factor_steering_angle" value="0.5" /-->

                <param name="vehicle_length" value="0.34" />
                <param name="max_steering_angle" value="0.52359877559" /> <!-- 30Â° -->
                <param name="max_steering_angle_speed" value="4.0" />

                <param name="look_ahead_dist" value="1.5" />
                <param name="k_o" value="0.0" />
                <param name="k_g" value="0.5" />
                <param name="k_w" value="0.0" />
                <param name="k_curv" value="0.0" />
                <param name="obst_threshold" value="3.0" />
            </group>
        </group>
    </group>  
</launch>
